execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -728,6 +729,10 @@ pub enum GenesisError {
         /// Number of validator slots specified.
         validator_slots: u32,
     },
+    /// Failed to push first item on runtime stack.
+    ///
+    /// This likely happened due to a misconfigured stack height.
+    StackTooSmall(RuntimeStackOverflow),
```

Comment:
There's no benefit to wrapping the `RuntimeStackOverflow` as it contains no additional info about the error - it's an empty struct.  I'd probably just change this to:
```suggestion
    StackTooSmall,
```